src/components/include/utils/threads/thread.h

```diff
@@ -114,6 +114,7 @@ class Thread {
   // Should be locked to protect isThreadRunning_ and thread_created_ values
   sync_primitives::Lock state_lock_;
   volatile bool isThreadRunning_;
+  volatile bool do_iteration_;
```

Comment:
@orhan-mehmedov we'are using a lot of variables for thread state describing (3 for now) and it's confusing. Could we use one variable with some custom enum values? For example `{STARTED, RUNNING, FINISHED, etc}`